tap_dayforce/__init__.py

```diff
@@ -72,24 +73,25 @@ def sync(args):
             singer.write_state(stream.state)
 
 
-def main():
+def _main():
     args = parse_args(required_config_keys={"username", "password", "client_namespace", "start_date"})
     if args.discover:
-        try:
-            discover(args, select_all=args.select_all)
-        except:
-            LOGGER.exception('Caught exception during Discovery..')
-            if log_to_rollbar is True:
-                rollbar.report_exc_info()
+        discover(args, select_all=args.select_all)
     elif not args.catalog:
         raise RuntimeError('Catalog file must be supplied during Sync.')
     else:
-        try:
-            sync(args)
-        except:
-            LOGGER.exception('Caught exception during Sync..')
-            if log_to_rollbar is True:
-                rollbar.report_exc_info()
+        sync(args)
+
+
+def main():
+    try:
+        _main()
+    except Exception:
+        if log_to_rollbar is True:
+            LOGGER.info("Reporting exception info to Rollbar..")
+            rollbar.report_exc_info()
+        LOGGER.critical(msg="Uncaught Exception..", exc_info=True)
```

Comment:
and now you can go back to the slightly less verbose and semantically agreeable:
```
LOGGER.exception("Uncaught Exception")
```